### DISCUSSION TITLE
How to read environment variables in pytest.ini

### DISCUSSION THREAD

**Discussion by 1Mark** [pytest-dev/pytest#9698] [Q&A]
Is there a way to read environment variables in `pytest.ini` ? [`pytest-env`](https://github.com/MobileDynasty/pytest-env) does the opposite and lets you set environment variables in there but my issue is different.

[`pytest-mongodb`](https://github.com/mdomke/pytest-mongodb#configuration) requires these three variables to be set in `pytest.ini`
```
[pytest]
mongodb_engine = pymongo
mongodb_host = mongodb://user:passwd@server.tld
mongodb_dbname = mydbname
```
However, I would like pytest.ini to read my environement variables instead.

```
[pytest]
mongodb_engine = $MONGO_ENGINE
mongodb_host = $MONGO_HOST
mongodb_dbname = $MONGO_DBNAME
```

Now obviously this isn't supported out the box.

Options:

1. Append to `pytestconfig.inicfg` but `pytestconfig` is a fixture and fixtures can't be used in hooks AFAIK. So it won't execute before `pytest-mongodb` begins - which will cause issues.
2. Raise a PR to `pytest-mongodb` which solves the issue for this repo but not for all the others using this annoying design.
3. Any other suggestions???
4. I use tox so if I invoke pytest via tox it solves the issue (via `passenv`) but this isn't how vscode operates with tests so it will be problematic and I won't be able just to use pytest raw on the CLI

**Comment by RonnyPfannschmidt** [comment]
pytest will NOT substutute env vars in the configfile

whatever uses the variables will have to do that on its own so i recommend 2.

as for passing credentials "propperly" to pytest, im currently not aware of a plugin that resolves it

**Reply by 1Mark** [reply]
So there isn't a programatic way to set the value for a variables in the pytest.ini early on?

**Reply by 1Mark** [reply]
I was considering https://docs.pytest.org/en/latest/example/markers.html#custom-marker-and-command-line-option-to-control-test-runs

**Reply by 1Mark** [reply]
I guess at least may be worth having a PR to mention that this isn't possible within pytest to avoid people wasting time :). As for pytest-mongodb I will try to migrate away from it since it seems it was abandoned since 2019 - or it's just perfect since 2019 but I assume not.

**Reply by 1Mark** [reply]
```python
def pytest_addoption(parser, pluginmanager):
    parser.addini('mongo_engine', help='pytest-mongodb requires the mongo engine', type='string', default='pymongo')
```
I did try this but it's not appending, 

in the test when I check the `pytestconfig.inicfg` it doesn't have the variable.

**Reply by 1Mark** [reply]
*I added the pytest_addoption in conftest.py

It is being invoked but the changes aren't persisting
